app/models/concerns/message_export_operations.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module MessageExportOperations
+  extend ActiveSupport::Concern
+
+  included do
+    def prepare_message_export
+      ::Zip::OutputStream.write_buffer do |zip|
+        prepare_original_objects(zip)
+        prepare_print_objects(zip)
+      end.string
+    end
+
+    private
+
+    def prepare_original_objects(zip)
+      objects.each do |message_object|
+        zip.put_next_entry("originaly/#{MessageObjectHelper.displayable_name(message_object)}")
+        zip.write(message_object.content)
+      end
+    end
+
+    def prepare_print_objects(zip)
+      objects.each do |message_object|
+        pdf_content = message_object.prepare_pdf_visualization
+
+        next unless pdf_content
```

Comment:
Nie je toto nebezpecne? Ak NEvieme vytvorit pdf vizualizaciu tak ju to zhltne?

Reply:
@luciajanikova toto este doriesme. imho by asi mal failnut export ak nieco failne